src/SoftekBarcodeAPI.php

```diff
@@ -37,8 +40,101 @@ public function __construct(SoftekBarcodeConfig $configuration)
 
             // license the instance for further operations
             $this->ffi->mtSetLicenseKey($this->instance, $configuration->getLicenseKey());
-        } catch (\Throwable $e) {
-            throw new SoftekInitializationException($e->getMessage(), "", $e);
+        } catch (Throwable $e) {
+            throw new SoftekInitializationException($e->getMessage(), 0, $e);
+        }
+    }
+
+    /**
+     * Scan the specified image file for bar code strings and return the detected barcodes in the file as Objects.
+     * Note that the function will stop when the first barcode is found in a document unless
+     * the MultipleRead property is set to True
+     * https://www.bardecode.com/en1/scanbarcode-scan-an-image-for-barcodes/
+     *
+     * @param string $pathToFile
+     * @return BarcodeScanResult[]
+     * @throws SoftekProcessImageException if an error occurs during extraction of the barcode from the InputFile
+     */
+    public function processImage(string $pathToFile): array
+    {
+        $result = $this->ffi->mtScanBarCode($this->instance, $pathToFile);
+        if ($result < 0) {
+            throw new SoftekProcessImageException($result, $pathToFile);
+        }
+
+        // if no barcodes detected return empty array
+        if ($result <= 0) return [];
+
+        return $this->extractScanResults($result);
+    }
+
+
+    /**
+     * This method extract the information using the barcode toolkit. It is required to run FFI::mtScanBarcode before
+     * this method to perform a legit request to the barcode toolkit interface.
+     *
+     * @param int $resultCount
+     * @return BarcodeScanResult[]
+     */
+    private function extractScanResults(int $resultCount): array
+    {
+        $barcodeScanResults = [];
+        for ($barcodeIndex = 1; $barcodeIndex <= $resultCount; $barcodeIndex++) {
+            $barcodeScanResult = new BarcodeScanResult();
+            // converting the cdate byte array type to a readable string
+            $barcodeScanResult
+                ->setText($this->ffiByteArrayToString($this->ffi->mtGetBarString($this->instance, $barcodeIndex)))
+                ->setType($this->ffiByteArrayToString($this->ffi->mtGetBarStringType($this->instance, $barcodeIndex)));
+
+            // switch type of variable to be confirmed with operating system
+            $ffiType = (PHP_OS === 'Linux') ? "long" : "int";
```

Comment:
Is it worth to have this in a `static function integerType(): string` ? May or may not be premature refactoring...

Reply:
Sound justified